ikarus/python/finiteelements/registerpreelement.hh

```diff
@@ -33,7 +33,8 @@ void registerNonLinearElasticPre(pybind11::handle scope, pybind11::class_<NonLin
  */
 template <class LinearElasticPre, class... options>
 void registerLinearElasticPre(pybind11::handle scope, pybind11::class_<LinearElasticPre, options...> cls) {
-  cls.def(pybind11::init([](double emod, double nu) { return new LinearElasticPre({emod, nu}); }));
+  using Material = typename LinearElasticPre::Material;
+  cls.def(pybind11::init([](const Material& mat) { return new LinearElasticPre(mat); }));
```

Comment:
To support all material parameter variants, we can add a constructor accepting `py::kwargs`, which will then basically be a dictionary. This dictionary we can the search for valid Pairs of materials  `LinearElasticity(E=1000, nu=0.2)` with "E" and "nu" as entries in the dictionary and checking for size of entries.

Reply:
For now I am doing the template argument decuction for linear elastic elmement same as nonlinearelastic  via the material like thaat
```python
def linearElastic(mat):
    """
    @brief Creates a linear elastic pre-element.

    @param mat: The material.

    @return: The registered linear elastic pre-element function.
    """
    includes = ["ikarus/finiteelements/mechanics/linearelastic.hh"]
    element_type = f"Ikarus::LinearElasticPre<{mat.cppTypeName}>"
    includes += mat._includes
    return registerPreElement("LinearElasticPre", includes, element_type, mat)
    ```
Therefore we kind of need the material in the constructor

For linear elastic we could make a default to LinearElasticity, but then we have to deduce the dimension and wether its plane strain or plane stress for 2D

We could rather change the constructor of the materials itself to be able to be constructed from all kind of material parameter combinations

Reply:
Like this 
```cpp
Ikarus::LamesFirstParameterAndShearModulus extractMaterialParameters(const pybind11::kwargs& kwargs) {
  if (kwargs.contains("E") and kwargs.contains("nu"))
    return Ikarus::toLamesFirstParameterAndShearModulus(
        {.emodul = kwargs["E"].cast<double>(), .nu = kwargs["nu"].cast<double>()});
  else if (kwargs.contains("E") and kwargs.contains("mu")) {
    auto converter =
        convertLameConstants(YoungsModulusAndShearModulus{kwargs["E"].cast<double>(), kwargs["mu"].cast<double>()});
    return {converter.toLamesFirstParameter(), kwargs["mu"].cast<double>()};
  }
}
```

Reply:
Haha unfortionaly we can not use "lambda" as a kwarg, as this is a reserved keyword in the python language 😆